File: core/src/main/java/cn/authing/core/http/Call.java
package cn.authing.core.http;


import org.jetbrains.annotations.Nullable;

import java.io.IOException;

import okhttp3.Request;

/**
 * copy from retrofit2
 */
public interface Call<T> {

    @Nullable T execute() throws IOException;

    void enqueue(Callback<T> callback);

    boolean isExecuted();

    void cancel();

    boolean isCanceled();

    Request request();
}

File: src/main/kotlin/cn/authing/core/http/Call.kt
package cn.authing.core.http

import okhttp3.Request
import java.io.IOException

interface Call<TData, TResult> {
    @Throws(IOException::class)
    fun execute(): TResult
    fun enqueue(callback: Callback<TResult>)
    fun isExecuted(): Boolean
    fun cancel()
    fun isCanceled(): Boolean
    fun request(): Request?
}